src/GraphQL/Utility/FileUpload.php

```diff
@@ -0,0 +1,232 @@
+use Drupal\Component\Utility\Bytes;
+use Drupal\Component\Utility\Environment;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\File\FileSystem;
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\graphql\Wrappers\FileUploadResponse;
+use Symfony\Component\HttpFoundation\File\MimeType\MimeTypeGuesserInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Service to manage file uploads within GraphQL mutations.
+ */
+class FileUpload {
+
+  use StringTranslationTrait;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The mime type guesser service.
+   *
+   * @var \Symfony\Component\HttpFoundation\File\MimeType\MimeTypeGuesserInterface
+   */
+  protected $mimeTypeGuesser;
+
+  /**
+   * The file system service.
+   *
+   * @var \Drupal\Core\File\FileSystemInterface
+   */
+  protected $fileSystem;
+
+  /**
+   * GraphQL logger channel.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelInterface
+   */
+  protected $logger;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entityTypeManager,
+    AccountProxyInterface $currentUser,
+    MimeTypeGuesserInterface $mimeTypeGuesser,
+    FileSystemInterface $fileSystem,
+    LoggerChannelInterface $logger
+  ) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->currentUser = $currentUser;
+    $this->mimeTypeGuesser = $mimeTypeGuesser;
+    $this->fileSystem = $fileSystem;
+    $this->logger = $logger;
+  }
+
+  /**
+   * Gets max upload size.
+   *
+   * @param array $settings
+   *   The file field settings.
+   *
+   * @return int
+   *   Max upload size.
+   */
+  protected function getMaxUploadSize(array $settings) {
+    // Cap the upload size according to the PHP limit.
+    $max_filesize = Bytes::toInt(Environment::getUploadMaxSize());
+    if (!empty($settings['max_filesize'])) {
+      $max_filesize = min($max_filesize, Bytes::toInt($settings['max_filesize']));
+    }
+    return $max_filesize;
+  }
+
+  /**
+   * Retrieves the upload validators for a file field.
+   *
+   * @param array $settings
+   *   The file field settings.
+   *
+   * @return array
+   *   List of file validators.
+   */
+  protected function getUploadValidators(array $settings) {
+    // Validate name length.
+    $validators = [
+      'file_validate_name_length' => [],
+    ];
+
+    // There is always a file size limit due to the PHP server limit.
+    $validators['file_validate_size'] = [$this->getMaxUploadSize($settings)];
+
+    // Add the extension check if necessary.
+    if (!empty($settings['file_extensions'])) {
```

Comment:
For security reasons we should have a default value of "txt" here if the user forgot to configure allowed file extensions by accident.